scanner/indexer/remote.go

```diff
@@ -0,0 +1,66 @@
+package indexer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/quay/claircore"
+	"github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/scanner/mappers"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// remoteIndexer is the Indexer implementation that connects to a remote indexer
+// using gRPC.
+type remoteIndexer struct {
+	conn    *grpc.ClientConn
+	indexer v4.IndexerClient
+}
+
+// NewRemoteIndexer connect to the gRPC address and creates a new remote indexer.
+func NewRemoteIndexer(ctx context.Context, address string) (*remoteIndexer, error) {
+	connOpt := clientconn.Options{
+		TLS: clientconn.TLSConfigOptions{
+			GRPCOnly:           true,
+			InsecureSkipVerify: true,
+		},
+	}
+	// TODO: [ROX-19050] Set the Scanner V4 TLS validation and the correct subject
+	//       when certificates are ready.
+	conn, err := clientconn.GRPCConnection(ctx, mtls.ScannerSubject, address, connOpt)
+	if err != nil {
+		return nil, err
+	}
+	return &remoteIndexer{
+		conn:    conn,
+		indexer: v4.NewIndexerClient(conn),
+	}, nil
+}
+
+// Close closes the remote indexer.
+func (r *remoteIndexer) Close(ctx context.Context) error {
+	return r.conn.Close()
+}
+
+// GetIndexReport call the remote service to retrieve an IndexReport for the given hash ID.
```

Comment:
```suggestion
// GetIndexReport calls the remote service to retrieve an IndexReport for the given hash ID.
```

Reply:
d8680a5141..7b472c4ee6

Reply:
Please, close if you don't have any other nipicks here.